File: templates/backend/django/levels/level_4/files/backend/app/services/todo_service.py
from app.repositories.todo_repository import TodoRepository


class TodoService:
    def __init__(self, repository: TodoRepository) -> None:
        self.repository = repository

    def list_todos(self) -> list[dict[str, object]]:
        return self.repository.list()

    def create_todo(self, title: str) -> dict[str, object]:
        return self.repository.create(title)

File: templates/backend/spring_boot/levels/level_4/files/backend/src/main/java/dev/reversevibe/backend/TodoService.java
package dev.reversevibe.backend;

import java.util.List;

class TodoService {
  private final TodoRepository repository;

  TodoService(TodoRepository repository) { this.repository = repository; }
  List<Todo> listTodos() { return repository.list(); }
  Todo createTodo(String title) { return repository.create(title); }
}
